### DISCUSSION TITLE
Can RLS Policy restrict a query based on select amount?

### DISCUSSION THREAD

**Discussion by rycont** [supabase/supabase#11001] [Questions] [answered]
I'd like to configure RLS Policies to allow anon users to select only one record per time. Can i build this policy with RLS?

**Comment by GaryAustin1** [comment, marked as answer]
I can't think of a way as RLS is done on a row by row basis.
You probably need to do a view or a security definer function rpc call that fetches one row.  For the rpc call you would then lock the table down so you can't do any select.  Then only way is thru the rpc call as it bypasses rls.

**Reply by rycont** [reply]
Thanks for your answer! I newly learned that Security Definer Function can bypass RLS Policy. It was perfect solution for my product.
